src/scripts/generate-binary-ops.py

```diff
@@ -0,0 +1,281 @@
+            else:
+                if (options.op[i] == "^"):
+                    self.operators_list.append(self.bitxor)
+        self.operands = [0] * num_operands
+        self.operators = [0] * (num_operands -1)
+
+        if (bit == False):
+            for i in range(num_operands -1):
+                self.operands[i] = random.randint(MIN, MAX)
+                self.operators[i] = random.randint(0, len(self.operators_list) -1)
+            self.operands[num_operands -1] = random.randint(MIN, MAX)
+
+        else:
+            for i in range(num_operands -1):
+                self.operands[i] = random.randint(BITMIN, BITMAX)
+                self.operators[i] = random.randint(0, len(self.operators_list) -1)
+            self.operands[num_operands -1] = random.randint(BITMIN, BITMAX)
+
+    def add(self, op1, op2):
+        sum = op1 + op2
+        if sum < MIN:
+            op2 = MIN - random.randint(MIN, op1)
+            sum = op1 + op2
+        elif sum > MAX:
+            op2 = MAX - random.randint(op1, MAX)
+            sum = op1 + op2
+        return {'oper': '+', 'op2': op2, 'sum': sum}
+
+
+    def sub(self, op1, op2):
+        sum = op1 - op2
+        if sum < MIN:
+            op2 = - (MIN - random.randint(MIN, op1))
+            sum = op1 - op2
+        elif sum > MAX:
+            op2 = - (MAX - random.randint(op1, MAX))
+            sum = op1 - op2
+        return {'oper': '-', 'op2': op2, 'sum': sum}
+
+    def mul(self, op1, op2):
+        sum = op1 * op2
+        if sum < MIN:
+            op2 = MIN / op1
+        elif sum > MAX:
+            op2 = MAX / op1
+        if op2 > 0:
+            op2 = math.floor(op2)
+        else:
+            op2 = math.ceil(op2)
+        sum = op1 * op2
+        return {'oper': '*', 'op2': op2, 'sum': sum}
+
+    def div(self, op1, op2):
+        if op2 == 0:
+            op2 = 2
+        sum = op1 / op2
+        if sum > 0:
+            sum = int(math.floor(sum))
+        else :
+            sum = int(math.ceil(sum))
+        return {'oper': '/', 'op2': op2, 'sum': sum}
+
+    def mod(self, op1, op2):
+        if op2 == 0:
+            op2 = 2
+        if (((op1 < 0 and op2 > 0) or (op1 > 0 and op2 < 0)) and (op1 % op2 != 0)):
+            sum = ((op1 % op2) - op2)
+        else:
+            sum = (op1 % op2)
+        return {'oper': '%', 'op2': op2, 'sum': sum}
+
+    def bitshiftleft(self, op1, op2):
+        op2 = int(math.fabs(op2))
+        if op1 < 0:
+            op2 = math.floor(math.log(MIN/op1))
+        else:
+            op2 = math.ceil(math.log(MIN/op1))
+        sum = op1 << op2
+        if sum < MIN:
+            if op1 < 0:
+                op2 = math.floor(math.log(MIN/op1, 2))
+            else:
+                op2 = math.ceil(math.log(MIN/op1, 2))
+        elif sum > MAX:
+            if op1 > 0:
+                op2 = math.floor(math.log(MAX/op1, 2))
+            else:
+                op2 = math.ceil(math.log(MAX/op1, 2))
```

Comment:
I'm still not sure why we have to round this value. Could you explain it?

Reply:
Because if it might be a float, and `sum` might jump out of the interval if `op2` is not rounded.